airflow/www/security.py

```diff
@@ -521,6 +524,18 @@ def sync_roles(self):
         self.update_admin_perm_view()
         self.clean_perms()
 
+    def sync_resource_permissions(self):
+        """
+        Populates resource-based permissions.
+        """
+        resources = ['Pool', 'Connection', 'Dag', 'DagBag', 'DagRun',
+                     'DagCode', 'Log', 'Task', 'ImportError', 'Variable', 'XCom']
+        actions = ['can_create', 'can_read', 'can_edit', 'can_delete']
```

Comment:
some of the resources here like ImportError doesn't have edit/delete/create endpoints right?

Reply:
Good point @houqp. I've updated it so that the resource is created dynamically when the decorator is called on the function.